server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -178,8 +196,19 @@ private MappedFieldType getDynamicField(String field) {
      * All field names in the returned set are guaranteed to resolve to a field
      */
     Set<String> getMatchingFieldNames(String pattern) {
```

Comment:
Having these two methods is really annoying me now :(

Reply:
Actually, thinking about it, maybe we can reimplement `getMatchingFieldTypes()` as a call to `getMatchingFieldNames()` and then a resolve on all values in the resulting set? With a shortcut for a match-all regex?

Reply:
I can give it a try. I tried the opposite and that did not fly :) but the other way around may indeed work